Sources/OpenAPIURLSession/URLSessionBidirectionalStreaming/BidirectionalStreamingURLSessionDelegate.swift

```diff
@@ -164,7 +164,8 @@ final class BidirectionalStreamingURLSessionDelegate: NSObject, URLSessionTaskDe
     {
         callbackLock.withLock {
             debug("Task delegate: didReceive response")
-            self.responseContinuation?.resume(returning: response)
+            responseContinuation?.resume(returning: response)
+            responseContinuation = nil
```

Comment:
I remember someone recently mentioning that we should be careful about resuming continuations within locked sections. I wonder if we could make this a little safer by returning the continuation from `withLock` and resuming it outside the locked section? Same below.

Still approving the PR, just a thought in case you haven't considered this.

Reply:
It's a good callout, but this lock we're holding here is used only for exclusivity between delegate callbacks, which are called by URLSession. The continuation for the user task here won't be calling these callbacks, and we immediately return after resuming anyway, which will drop the lock.